Desktop/data/datasetpackage.cpp

```diff
@@ -2530,7 +2530,7 @@ void DataSetPackage::setAnalysesData(const Json::Value &analysesData)
 {
 	QString		previousASF					= analysesData.type() != Json::objectValue ? "" : tq(analysesData.get("autoSaveFileName", "").asString());
 				_analysesData				= analysesData;
-	QFileInfo	dataFile					( tq(_dataSet->dataFilePath()) ),
+	QFileInfo	dataFile					( tq(_dataSet ? _dataSet->dataFilePath() : "") ),
```

Comment:
Shouldnt there always be an empty DataSet anyway?

Reply:
And you are right: if you close the workspace, add a an analysis without data, and save manually, it crashes: no need to wait for the autosave.

Reply:
OK, then I change the code so that DataSetPackage always has a proper DataSet object